common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockHandler.java

```diff
@@ -290,6 +290,10 @@ public ShuffleMetrics() {
       final int[] mapIdAndReduceIds = new int[2 * blockIds.length];
       for (int i = 0; i < blockIds.length; i++) {
         String[] blockIdParts = blockIds[i].split("_");
+        if (blockIdParts.length == 5) {
```

Comment:
the `if` below can fail as well, do we need this check?

Reply:
Yes, not so useful, just want to keep the same exception message, let me revert.